### PR TITLE
test: add no-extra-parens tests with rest properties

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[x] Other, please explain:

Uncomments two `no-extra-parens` tests from https://github.com/eslint/eslint/pull/12490. The tests were commented out because parser didn't allow parentheses around the assignment target in rest properties. This has recently been fixed in Acorn (Espree PR https://github.com/eslint/espree/pull/554).

#### What changes did you make? (Give an overview)

Uncommented the tests.

### CODE REVIEW

**Comment by mdjermanovic** [comment]
I'll close-reopen to see if that will update the CodeQL check.

**Comment by mdjermanovic** [comment]
The close-reopen didn't seem to update the CodeQL check, but the reported problem is completely unrelated to this PR and has already been fixed in the `main` branch, so I think it's safe to merge this once it's approved.

**Comment by nzakas** [comment]
You’ll need to rebase in order to clear the CodeQL warning. I had to do the same in another PR.

**Review by nzakas — APPROVED** [review]

**Comment by mdjermanovic** [comment]
> You’ll need to rebase in order to clear the CodeQL warning. I had to do the same in another PR.

It worked, thanks!

**Pull request merged**
